plugins/formal-verification/formver.viper/src/org/jetbrains/kotlin/formver/viper/ast/Position.kt

```diff
@@ -8,12 +8,33 @@ package org.jetbrains.kotlin.formver.viper.ast
 import org.jetbrains.kotlin.formver.viper.IntoSilver
 import viper.silver.ast.`NoPosition$`
 
+data class PositionWrapper<P>(val ktPosition: P) : viper.silver.ast.Position
```

Comment:
Why do you need separate classes for `PositionWrapper` and `KtSourcePosition`?  In fact, it isn't quite clear to me that the code here needs to care about it at all: I think you could provide an open class `Position.Wrapped` (or something) that inherits from `viper.silver.ast.Position` and that the user can inherit from in their code; I don't see where `formver.viper` needs to be aware of what kind of positions we have.